src/krux/pages/tools.py

```diff
@@ -173,3 +174,16 @@ def create_qr(self):
             seed_qr_view = SeedQRView(self.ctx, data=text, title=title)
             return seed_qr_view.display_qr(allow_export=True)
         return MENU_CONTINUE
+
+    def wallet_addresses(self):
+        """Handler for the 'Wallet Addresses' menu item"""
+        from .home_pages.wallet_descriptor import WalletDescriptor
+        from .home_pages.addresses import Addresses
+        from krux.wallet import Wallet
+
+        if not self.ctx.wallet:
```

Comment:
note: Tadeu and I discussed that preserving the wallet while not yet logged in (in tools) doesn't feel right, this line to be removed so that reviewing full descriptor and qrcode is not possible until actually logged in with a real HDKey.